src/data_structures/linked_list.md

```diff
@@ -0,0 +1,25 @@
+### Linked list
+
+Almost identical to the Array list . They share the same functions due to them both sharing the `List` interface. Unlike ArrayList it doesn't Have an Array fundamentally though. it's a series of private nested classes of type `Entry` it references a `next` and `previous` item. Its like containers linked to each other.
```

Comment:
i don't see any compelling reason to introduce this. Linked lists are useful for teaching recursion and interfaces, but this particular class isn't really useful...ever.